hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -339,159 +227,176 @@ public void noMoreTransactions() {
-		return getChild(ChildIndices.STORAGE);
+		return getChild(StateChildIndices.STORAGE);
 	}
 
 	public FCMap<MerkleEntityId, MerkleTopic> topics() {
-		return getChild(ChildIndices.TOPICS);
+		return getChild(StateChildIndices.TOPICS);
 	}
 
 	public FCMap<MerkleEntityId, MerkleToken> tokens() {
-		return getChild(ChildIndices.TOKENS);
+		return getChild(StateChildIndices.TOKENS);
 	}
 
 	public FCMap<MerkleEntityAssociation, MerkleTokenRelStatus> tokenAssociations() {
-		return getChild(ChildIndices.TOKEN_ASSOCIATIONS);
+		return getChild(StateChildIndices.TOKEN_ASSOCIATIONS);
 	}
 
 	public FCMap<MerkleEntityId, MerkleSchedule> scheduleTxs() {
-		return getChild(ChildIndices.SCHEDULE_TXS);
+		return getChild(StateChildIndices.SCHEDULE_TXS);
 	}
 
 	public MerkleNetworkContext networkCtx() {
-		return getChild(ChildIndices.NETWORK_CTX);
+		return getChild(StateChildIndices.NETWORK_CTX);
 	}
 
 	public AddressBook addressBook() {
-		return getChild(ChildIndices.ADDRESS_BOOK);
+		return getChild(StateChildIndices.ADDRESS_BOOK);
 	}
 
 	public MerkleDiskFs diskFs() {
-		return getChild((ChildIndices.DISK_FS));
+		return getChild((StateChildIndices.DISK_FS));
 	}
 
 	public RecordsRunningHashLeaf runningHashLeaf() {
-		return getChild(ChildIndices.RECORD_STREAM_RUNNING_HASH);
+		return getChild(StateChildIndices.RECORD_STREAM_RUNNING_HASH);
 	}
 
 	public FCMap<MerkleUniqueTokenId, MerkleUniqueToken> uniqueTokens() {
-		return getChild(ChildIndices.UNIQUE_TOKENS);
+		return getChild(StateChildIndices.UNIQUE_TOKENS);
 	}
 
 	public FCOneToManyRelation<Integer, Long> uniqueTokenAssociations() {
-		return uniqueTokenAssociations;
+		return metadata.getUniqueTokenAssociations();
 	}
 
 	public FCOneToManyRelation<Integer, Long> uniqueOwnershipAssociations() {
-		return uniqueOwnershipAssociations;
+		return metadata.getUniqueOwnershipAssociations();
 	}
 
 	public FCOneToManyRelation<Integer, Long> uniqueTreasuryOwnershipAssociations() {
-		return uniqueTreasuryOwnedAssociations;
+		return metadata.getUniqueTreasuryOwnershipAssociations();
 	}
 
-	void setUniqueTokenAssociations(FCOneToManyRelation<Integer, Long> uniqueTokenAssociations) {
-		this.uniqueTokenAssociations = uniqueTokenAssociations;
+	private void internalInit(Platform platform, BootstrapProperties bootstrapProps) {
+		setImmutable(false);
+
+		networkCtx().setStateVersion(StateVersions.CURRENT_VERSION);
+		diskFs().checkHashesAgainstDiskContents();
+
+		final var selfId = platform.getSelfId();
+		ServicesContext ctx;
+		if (CONTEXTS.isInitialized(selfId.getId())) {
+			ctx = CONTEXTS.lookup(selfId.getId());
+		} else {
+			final var properties = new StandardizedPropertySources(bootstrapProps);
+			ctx = new ServicesContext(selfId, platform, this, properties);
+		}
+		metadata = new StateMetadata(ctx);
+
+		contextInitializer.accept(this, ctx);
+		CONTEXTS.store(ctx);
+		logSummary();
+
+		log.info("  --> Context initialized accordingly on Services node {}", selfId);
```

Comment:
Not sure the logging pattern hedera uses. In Swirlds we always use a log marker, not sure if one should also be used here.

Reply:
We have a TODO to adopt markers, but haven't atm. :/